scripts/cicd/test_syslog_monitor_memory_usage/run_test.sh

```diff
@@ -0,0 +1,105 @@
+#! /bin/bash
+
+TCP_SERVERS=$1
+UDP_SERVERS=$2
+INTERVAL=$3
+
+echo_with_date() {
+    date +"[%Y-%m-%d %H:%M:%S] $*"
+}
+
+stop_agent() {
+  python scalyr_agent/agent_main.py --config $CONFIG_FILE stop
+}
+
+trap stop_agent EXIT
+
+function generate_config() {
+  TCP_SERVERS=$1
+  UDP_SERVERS=$2
+
+  echo "{" > agent.json
+
+  echo '
+"import_vars": ["SCALYR_API_KEY"],
+"scalyr_server": "agent.scalyr.com",
+"api_key": "$SCALYR_API_KEY",
+"log_rotation_max_bytes": "1073741824",
+"syslog_processing_thread_count": "16",
+
+"server_attributes": {
+   "serverHost": "github-action-memory-test"
+
+},' >> agent.json
+
+echo "\"monitors\": [" >> agent.json
+
+for N in `seq $TCP_SERVERS`;
+do
+  echo "{
+      \"module\":                    \"scalyr_agent.builtin_monitors.syslog_monitor\",
+      \"protocols\":                 \"tcp:$(($N+1600))\",
+      \"accept_remote_connections\": \"true\",
+      \"message_log\": \"tcp_$N.log\"
+    }," >> agent.json
+done
+
+for N in `seq $UDP_SERVERS`;
+do
+  echo "{
+      \"module\":                    \"scalyr_agent.builtin_monitors.syslog_monitor\",
+      \"protocols\":                 \"udp:$(($N+1500))\",
+      \"accept_remote_connections\": \"true\",
+      \"message_log\": \"udp_$N.log\"
+    }," >> agent.json
+done
+
+echo "]" >> agent.json
+
```

Comment:
Using a here doc might be easier for writing the agent.json file

Reply:
Good idea!